src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java

```diff
@@ -189,29 +202,38 @@ public Object getConvertedValue(FacesContext context, UIComponent component, Obj
                 contentType = resource.getContentType();
             }
             else {
+                if (imagePath != null) {
 
-                boolean isExternal = imagePath.startsWith("http");
+                    boolean isExternal = imagePath.startsWith("http");
 
-                if (isExternal) {
-                    URL url = new URL(imagePath);
-                    URLConnection urlConnection = url.openConnection();
-                    inputStream = urlConnection.getInputStream();
-                    contentType = urlConnection.getContentType();
+                    if (isExternal) {
+                        URL url = new URL(imagePath);
+                        URLConnection urlConnection = url.openConnection();
+                        inputStream = urlConnection.getInputStream();
+                        contentType = urlConnection.getContentType();
+                    }
+                    else {
+                        ExternalContext externalContext = context.getExternalContext();
+                        // GitHub #3268 OWASP Path Traversal
+                        imagePath = FileUploadUtils.checkPathTraversal(imagePath);
+
+                        String webRoot = externalContext.getRealPath(Constants.EMPTY_STRING);
+                        String fileSeparator = Constants.EMPTY_STRING;
+                        if (!(webRoot.endsWith("\\") || webRoot.endsWith("/")) &&
+                                    !(imagePath.startsWith("\\") || imagePath.startsWith("/"))) {
+                            fileSeparator = "/";
+                        }
+
+                        File file = new File(webRoot + fileSeparator + imagePath);
+                        inputStream = new FileInputStream(file);
+                    }
+                }
+                else if (stream != null) {
+                    inputStream = stream.getStream();
+                    contentType = stream.getContentType();
                 }
                 else {
-                    ExternalContext externalContext = context.getExternalContext();
-                    // GitHub #3268 OWASP Path Traversal
-                    imagePath = FileUploadUtils.checkPathTraversal(imagePath);
-
-                    String webRoot = externalContext.getRealPath(Constants.EMPTY_STRING);
-                    String fileSeparator = Constants.EMPTY_STRING;
-                    if (!(webRoot.endsWith("\\") || webRoot.endsWith("/")) &&
-                                !(imagePath.startsWith("\\") || imagePath.startsWith("/"))) {
-                        fileSeparator = "/";
-                    }
-
-                    File file = new File(webRoot + fileSeparator + imagePath);
-                    inputStream = new FileInputStream(file);
+                    throw new IllegalArgumentException("Either image and stream attributes are not provided.");
```

Comment:
"Either image or stream attribute is not provided"

Reply:
Fixed!